### DISCUSSION TITLE
What is the cost of transferring data from Go to JS?

### DISCUSSION THREAD

**Discussion by brentmparker** [wailsapp/wails#1723] [Q&A]
Say I have a moderately-sized slice of structs that I want to filter (like a fuzzy-matching search as someone types into an input). Would it be better to just transfer the entire slice to the frontend and do the filtering in JS or to send the query string to the backend, have Go do the filtering, and only return the results to the frontend?

What is the cost of translating data from go to JS? Is it pretty high? Do you happen to have any benchmarks?

**Comment by leaanthony** [comment]
It's a subjective question and the only answer is "try it and see". There are 3 ways to do it:
* make a call
* fire an event
* Use a custom AssetHandler
